packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/simulationservice/interventions/InterventionPolicy.java

```diff
@@ -40,4 +42,45 @@ public InterventionPolicy clone() {
 		}
 		return clone;
 	}
+
+	// Check each intervention this policy contains
+	// If any of them are are invalid the entire policy is invalid.
+	public Boolean validateInterventionPolicy() throws Exception {
+		for (int i = 0; i < this.interventions.size(); i++) {
+			this.interventions.get(i).validateIntervention(); //validate individual intervention
+			for (int j = 0; j < this.interventions.size(); j++) {
+				if (Integer.compare(i, j) != 0) {
+					validateInterventionPair(this.interventions.get(i), this.interventions.get(j)); //validate pair with eachother
+				}
+			}
+		}
+		return true;
+	}
+
+	// Takes a list of two interventions and will check their static intervention lists to ensure
+	// there are no duplicate time + appliedTo pairs in their indiviudal static interventions
+	private Boolean validateInterventionPair(Intervention interOne, Intervention interTwo) throws Exception {
```

Comment:
This seem to suggest you want either
- encode uniqueness through overriding toString or hasCode functions, or
- implement the Comparable interface and use compareTo(...)